providers/websockets/gate/parse.go

```diff
@@ -0,0 +1,70 @@
+package gate
+
+import (
+	"fmt"
+	"math/big"
+	"time"
+
+	"go.uber.org/zap"
+
+	"github.com/skip-mev/slinky/pkg/math"
+	"github.com/skip-mev/slinky/providers/base/websocket/handlers"
+	providertypes "github.com/skip-mev/slinky/providers/types"
+	oracletypes "github.com/skip-mev/slinky/x/oracle/types"
+)
+
+// parseSubscribeResponse attempts to parse a SubscribeResponse to see if it was successful.
+func (h *WebsocketDataHandler) parseSubscribeResponse(
+	msg SubscribeResponse,
+) ([]handlers.WebsocketEncodedMessage, error) {
+	if msg.Error.Message != "" {
+		errCode := ErrorCode(msg.Error.Code)
+		h.logger.Error("found error in subscribe response", zap.Error(errCode.Error()))
+		return nil, errCode.Error()
+	}
+
+	if Status(msg.Result.Status) != StatusSuccess {
+		h.logger.Error("subscription was not successful", zap.String("status", msg.Result.Status), zap.String("pair", msg.Error.Message))
+		return nil, fmt.Errorf("subscription was not successful: %s", msg.Result.Status)
+	}
```

Comment:
should we send a new sub message?

Reply:
We don't know which subscription failed in this case.  But then submitting a new subscription for all pairs (naive solution) will return a failure because we are already subbed.